src/Sylius/Bundle/OrderBundle/NumberGenerator/SequentialOrderNumberGenerator.php

```diff
@@ -92,11 +123,23 @@ private function getSequence()
         /** @var OrderSequenceInterface $sequence */
         $sequence = $this->sequenceRepository->findOneBy([]);
 
-        if (null === $sequence) {
-            $sequence = $this->sequenceFactory->createNew();
-            $this->sequenceRepository->add($sequence);
+        if (null !== $sequence) {
+            return $sequence;
         }
 
+        $sequence = $this->sequenceFactory->createNew();
+        $this->sequenceManager->persist($sequence);
+
         return $sequence;
     }
+
+    private function openSequenceManagerIfClosed()
+    {
+        if (!$this->sequenceManager->isOpen()) {
+            $this->sequenceManager = EntityManager::create(
```

Comment:
Yup, but doctrine doesn't give us possibility to reopen entity manager, we can only creating a new one, which is a solution recommended by doctrine documentation. Even `resetManager()` method on doctrine `Registry` creates a new manager. 
I will change the method name.